cadence/activity_loop.py

```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
                 )
-                _, err = service.respond_activity_task_completed(respond)
-                if err:
-                    logger.error("Error invoking RespondActivityTaskCompleted: %s", error)
+                service.respond_activity_task_completed(response)
             except Exception as ex:
-                logger.error(f"Activity {task.activity_type.name} failed: {type(ex).__name__}({ex})",exc_info=1)
-                respond: RespondActivityTaskFailedRequest 
-                respond = RespondActivityTaskFailedRequest(
+                logger.error(
+                    f"Activity {task.activity_type.name} failed: {type(ex).__name__}({ex})",
+                    exc_info=1)
+                response: RespondActivityTaskFailedRequest
+                response = RespondActivityTaskFailedRequest(
                     task_token=task.task_token,
                     reason="SOMTHING WENT WRONG",
                     identity=WorkflowService.get_identity(),
                     details=json.dumps({
-                        "detailMessage": f"Python error: {type(ex).__name__}({ex})"
-                    })
-                )
-                _, error = service.respond_activity_task_failed(respond)
-                if error:
-                    logger.error("Error invoking RespondActivityTaskFailed: %s", error)
-                    
+                        "detailMessage":
+                        f"Python error: {type(ex).__name__}({ex})"
+                    }))
+                try:
+                    service.respond_activity_task_failed(response)
+                except Exception as ex:
+                    logger.error("Error invoking RespondActivityTaskFailed: %s", ex)
+                    continue
+
             logger.info("Process ActivityTask: done")
-            
+    except Exception as ex:
+        logger.fatal(f"activity_task_loop: Uncought exception in main loop.",exec_info=True)
     finally:
         worker.notify_thread_stopped()
 
 
-def handle_task(task, activities):
+
+
+
+def handle_task(task, activities) -> Dict:
+    def handle_error(msg: str, exception: Type[Exception]) -> NoReturn:
+        logger.error(msg)
+        raise exception(msg)
+
     # TODO: finish this one
-    args = json.loads(task.input)
-    if not args:
-        # TODO: through acception
-    # fn = worker.activities.get(task.activity_type.name)
-    # if not fn:
-    #     logger.error("Activity type not found: " + task.activity_type.name)
-        
-    # return fn(*args)
-    return
-
-
-def handle_task_err(task):
-    return
+    if not task.task_token:
+        handle_error("task.task_token was not provided, but is expected.",
+                     RuntimeError)
+    if task.activity_type.name not in activities:
+        handle_error(
+            f"handle_task: Activity type {task.activity_type.name} not found",
+            RuntimeError)
+    fn = activities[task.activity_type.name]
```

Comment:
move it to line before 107

Reply:
We just checked that it exists the lines before. I think it's more clear if it is right when we have verified that it exists

Reply:
sure!